contracts/LockDealNFT/LockDealNFT.sol

```diff
@@ -94,15 +94,18 @@ contract LockDealNFT is LockDealNFTModifiers, ILockDealNFTEvents {
         address newOwner
     ) external onlyOwnerOrAdmin(poolId) {
         address provider = poolIdToProvider[poolId];
-        uint256 newPoolId;
+        uint256 dataPoolId;        
         // refund provider case
         if(poolIdToVaultId[poolId] == 0 && !approvedProviders[msg.sender]) {
-            IProvider(provider).split(poolId, 0, splitAmount);
+            newOwner = provider;
+            dataPoolId = poolId - 2;
         }
```

Comment:
I don't like the if here. need more info to understand it

Reply:
All `RefundProvider` pools are defined.
**poolId** - user pool ID
**(poolId - 1)** is always the ID of the refund data pool.
**(poolId - 2)** is always the user data pool ID.

When we use split, mint will always come first, we need to set it to the provider's split function. If we set not first we destroy `RefundProvider` logic.  So it will be **poolId - 2** in the `RefundProvider`. The owner **(poolId - 2)** will be the address of the provider instead of newOwner. We must make the correct display of data in the pools.